Python/Product/Analysis/LanguageServer/Settings.cs

```diff
@@ -15,24 +15,62 @@
 // permissions and limitations under the License.
 
 using System;
+using System.Collections.Generic;
 
 namespace Microsoft.PythonTools.Analysis.LanguageServer {
-    internal class LanguageServerSettings {
-        public AutoCompleteSettings autoComplete;
-        public DiagnosticsSettings diagnostics;
+    public sealed class LanguageServerSettings {
+        public class PythonAnalysisOptions {
+            public bool openFilesOnly;
+            public object[] errors = Array.Empty<object>();
+            public object[] warnings = Array.Empty<object>();
+            public object[] information = Array.Empty<object>();
+            public object[] disabled = Array.Empty<object>();
 
-        [Serializable]
-        internal class AutoCompleteSettings {
-            public bool showAdvancedMembers;
-        }
+            private HashSet<object> _errors;
+            private HashSet<object> _warnings;
+            private HashSet<object> _information;
+            private HashSet<object> _disabled;
+
+            public DiagnosticSeverity GetEffectiveSeverity(object code, DiagnosticSeverity defaultSeverity) {
+                Init();
+
+                if (_disabled != null && _disabled.Contains(code)) {
+                    return DiagnosticSeverity.Unspecified;
+                }
+                if (_errors != null && _errors.Contains(code)) {
+                    return DiagnosticSeverity.Error;
+                }
+                if (_warnings != null && _warnings.Contains(code)) {
+                    return DiagnosticSeverity.Warning;
+                }
+                if (_information != null && _information.Contains(code)) {
+                    return DiagnosticSeverity.Information;
+                }
+                return defaultSeverity;
+            }
 
-        [Serializable]
-        internal class DiagnosticsSettings {
-            public DiagnosticSeverity unresolvedImports;
+            public bool Show(DiagnosticSeverity severity) => severity != DiagnosticSeverity.Unspecified;
```

Comment:
Is this really useful? Being a non-static method implies that some calculation is going on, but that's really not the case, so we may as well just check it where we are decided to hide the entire message.